assignments/prototype-refactor.js

```diff
@@ -7,3 +7,197 @@ Prototype Refactor
+  
+  Each constructor function has unique properites and methods that are defined in their block comments below:
+*/
+  
+/*
+  === GameObject ===
+  * createdAt
+  * dimensions
+  * destroy() // prototype method -> returns the string: 'Object was removed from the game.'
+*/
+
+
+
+
+/*const GameObject = function (attributes){
+    this.createdAt = attributes.createdAt;
+    this.dimensions = attributes.dimensions;
+  }
+  GameObject.prototype.destroy = function() {
+    return `${this.name} was removed from the game.`;
+  }*/
+  /*
+    === CharacterStats ===
+    * hp
+    * name
+    * takeDamage() // prototype method -> returns the string '<object name> took damage.'
+    * should inherit destroy() from GameObject's prototype
+  */
+
+
+
+
+/*
+  const CharacterStats = function(attributes){
+    GameObject.call(this, attributes);
+    this.hp = attributes.hp;
+    this.name = attributes.name;
+  }
+  CharacterStats.prototype = Object.create(GameObject.prototype);
+  CharacterStats.prototype.takeDamage = function() {
+    return `${this.name} took damage.`
+  };
+  */
+  /*
+    === Humanoid ===
+    * faction
+    * weapons
+    * language
+    * greet() // prototype method -> returns the string '<object name> offers a greeting in <object language>.'
+    * should inherit destroy() from GameObject through CharacterStats
+    * should inherit takeDamage() from CharacterStats
+  */
+
+
+
+
+ /*
+   const Humanoid = function(attributes){
+     CharacterStats.call(this, attributes);
+     this.faction = attributes.faction;
+     this.weapons = attributes.weapons;
+     this.language = attributes.language;
+   }
+   
+   Humanoid.prototype = Object.create(CharacterStats.prototype);
+   
+   Humanoid.prototype.greet = function(){
+     return `${this.name} offers a greeting in ${this.language}.`
+   };
+  */
+  /*
+    * Inheritance chain: GameObject -> CharacterStats -> Humanoid
+    * Instances of Humanoid should have all of the same properties as CharacterStats and GameObject.
+    * Instances of CharacterStats should have all of the same properties as GameObject.
+  */
+  
+  // Test you work by uncommenting these 3 objects and the list of console logs below:
+  class GameObject{
+    constructor(attributes){
+    this.createdAt = attributes.createdAt;
+    this.dimensions = attributes.dimensions;
+    }
+    destroy(){
+        return `${this.name} was removed from the game.`;
+    }
+}
+
+
+class CharacterStats extends GameObject{
+    constructor(characterAttributes){
+        super(characterAttributes);
+        this.hp = characterAttributes.hp;
+        this.name = characterAttributes.name;
+    }
+    takeDamage(){
+      return `${this.name} took damage.`;
+    }
+}
+
+class Humanoid extends CharacterStats{
+    constructor(humanoidAttributes){
+      super(humanoidAttributes);
+      this.faction = humanoidAttributes.faction;
+      this.weapons = humanoidAttributes.weapons;
+      this.language = humanoidAttributes.language;
```

Comment:
This could be a candidate for destructuring as well:
```javascript
    constructor(humanoidAttributes) {
      const { faction, weapons, language } = humanoidAttributes;
      super(humanoidAttributes);
      this.faction = faction;
      this.weapons = weapons;
      this.language = language;
    }
```